plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/partition/HudiPartitionSplitGenerator.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi.partition;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+import io.trino.plugin.hive.HivePartitionKey;
+import io.trino.plugin.hudi.HudiSplit;
+import io.trino.plugin.hudi.HudiTableHandle;
+import io.trino.plugin.hudi.HudiUtil;
+import io.trino.plugin.hudi.split.HudiSplitWeightProvider;
+import io.trino.spi.connector.ConnectorSplit;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.HoodieTimer;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.exception.HoodieIOException;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HudiPartitionSplitGenerator
+        implements Runnable
+{
+    private static final Logger LOG = Logger.get(HudiPartitionSplitGenerator.class);
+    private final FileSystem fileSystem;
+    private final HoodieTableMetaClient metaClient;
+    private final HudiTableHandle tableHandle;
+    private final HudiSplitWeightProvider hudiSplitWeightProvider;
+    private final Map<String, HudiPartitionInfo> partitionInfoMap;
+    private final ArrayDeque<Pair<FileStatus, String>> hoodieFileStatusQueue;
+    private final ArrayDeque<ConnectorSplit> connectorSplitQueue;
+    private boolean isRunning;
+
+    public HudiPartitionSplitGenerator(
```

Comment:
Love to review this overall design/runtime of the split generator - ensure I understand all the batching and so on. Currently, this machinery feels a bit heavy for what we are accomplishing